packages/compiler/core/checker.ts

```diff
@@ -2226,21 +2224,20 @@ export function createChecker(program: Program): Checker {
     node: ModelExpressionNode | ModelStatementNode,
     properties: Map<string, ModelProperty>,
     parentModel: Model,
-    mapper: TypeMapper | undefined,
-    inheritedProps?: ModelProperty[]
+    mapper: TypeMapper | undefined
   ) {
     for (const prop of node.properties!) {
       if ("id" in prop) {
         const newProp = checkModelProperty(prop, mapper, parentModel);
         checkPropertyCompatibleWithIndexer(parentModel, newProp, prop);
-        defineProperty(properties, newProp, inheritedProps);
+        defineProperty(properties, newProp, parentModel);
```

Comment:
```suggestion
        defineProperty(properties, newProp);
```

This was a mistake by me, fixing.